lib/icinga/checkable-notification.cpp

```diff
@@ -229,7 +237,7 @@ bool Checkable::NotificationReasonApplies(NotificationType type)
 		case NotificationRecovery:
 			{
 				auto cr (GetLastCheckResult());
-				return cr && IsStateOK(cr->GetState());
+				return cr && (IsStateOK(cr->GetState()) || GetStateType() == StateTypeSoft);
```

Comment:
I think this no longer really matches the documented purpose of this function.

This function is also used in `notificationcomponent.cpp`, is this change also safe/desired for the use there?

Reply:
It does. "sending a notification of type type right now would represent *this' current state correctly" Hard OK -> soft crit is still hard OK. IMAO we could notify about the recovery.

Reply:
> Hard OK -> soft crit is still hard OK. IMAO we could notify about the recovery.

Isn't that the very same discussion as https://github.com/Icinga/icinga2/pull/9146#pullrequestreview-844828578? I wouldn't say "is still hard OK", they very nature of a soft state is that we're neither OK nor in a hard state.

> This function is also used in `notificationcomponent.cpp`, is this change also safe/desired for the use there?

Also, have you checked/tested this?

Reply:
Hmm.. if it still applies, it’s not subtracted.. if it’s not subtracted, it’s fired out.. any objections if I'm defensive and more or less dup c2a454306ff36fbcdde021d69e1c12c754e46c1a there? (s/continue/return/)

Reply:
If it's needed there, add it as well. What would happen there without this PR, i.e. does this change behavior due to the changed semantics of `NotificationReasonApplies`? If so, maybe we can just keep that function as-is and handle everything at the one affected call-site?